src/cow-react/common/pure/CurrencyInputPanel/types.ts

```diff
@@ -3,12 +3,12 @@ import { Currency, CurrencyAmount } from '@uniswap/sdk-core'
 import { ReceiveAmountInfo } from '@cow/modules/swap/helpers/tradeReceiveAmount'
 
 export interface CurrencyInfo {
-  field: Field
   label?: string
-  viewAmount: string
+  field: Field
+  currency: Currency | null
   rawAmount: CurrencyAmount<Currency> | null
+  viewAmount: string // TODO: should be calculated in CurrencyInputPanel
```

Comment:
I don't get the TODO. Also, you just re-ordered the fields, right? i didn't notice any change

Reply:
https://github.com/cowprotocol/cowswap/pull/2404
Here is a fix